designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,544 @@
+    1. ESLint expands the glob pattern to get a list of files.
+    1. Each file is checked individually as in step 1.
+1. When a directory is passed directly (such as `eslint src`):
+    1. The directory is converted into a glob pattern (such as `src` becomes `src/**`).
+    1. The glob pattern is checked as in step 2.
+
+## Documentation
+
+This will require extensive documentation changes and an introductory blog post.
+
+At a minimum, these pages will have to be updated (and rewritten):
+
+* https://eslint.org/docs/user-guide/getting-started#configuration
+* https://eslint.org/docs/user-guide/configuring
+* https://eslint.org/docs/developer-guide/working-with-plugins#configs-in-plugins
+* https://eslint.org/docs/developer-guide/shareable-configs
+
+## Drawbacks
+
+As with any significant change, there are some significant drawbacks:
+
+1. We'd need to do a phased rollout (see Backwards Compatibility Analysis) to minimize impact on users. That means maintaining two configuration systems for a while, increasing maintenance overhead and complexity of the application.
+1. Getting everyone to convert to `.eslint.js` format would be a significant stress on the community that could cause some resentment.
+1. We can no longer enforce naming conventions for plugins and shareable configs. This may make it more difficult for users to find compatible npm packages.
+1. Creating configuration files will be more complicated.
+1. The `--print-config` option becomes less useful because we can't output objects in a meaningful way.
+1. People depending on environments may find this change particularly painful.
+
+## Backwards Compatibility Analysis
+
+The intent of this proposal is to replace the current `.eslintrc` format, but can be implemented incrementally so as to cause as little disturbance to ESLint users as possible
+
+In the first phase, I envision this:
+
+1. `.eslint.js` can be implemented alongside `.eslintrc`.
+1. If `.eslint.js` is found, then:
+    1. All `.eslintrc` files are ignored.
+    1. `.eslintignore` is ignored.
+    1. `--ignore-file` is ignored.
+    1. `--rulesdir` is ignored.
+    1. `--env` is ignored.
+    1. `eslint-env` config comments are ignored.
+1. If `.eslint.js` is not found, then fall back to the current behavior.
+
+This keeps the current behavior for the majority of users while allowing some users to test out the new functionality. Also, `-c` could not be used with `.eslint.js` in this phase.
+
+In the second phase (and in a major release), ESLint will emit deprecation warnings whenever the original functionality is used but will still honor them so long as `.eslint.js` is not found.
+
+In the third phase (and in another major release), the original functionality is removed.
+
+So while this is intended to be a breaking change, it will be introduced over the course of three major releases in order to give users ample time to transition.
+
+## Alternatives
```

Comment:
Bikesheddable: A name like `eslint.config.js` rather than `.eslint.js` might make it clearer that the file is a config file. It would also avoid the issue where the config file is ignored by default due to being a dotfile.

Reply:
Changed to `.eslint.config.js` per your and @kaicataldo's comments.